client_taxi_driver/shared/src/commonMain/kotlin/presentation/map/MapScreenModel.kt

```diff
@@ -0,0 +1,98 @@
+package presentation.map
+
+import domain.entity.Order
+import domain.usecase.ManageOrderUseCase
+import presentation.base.BaseScreenModel
+import presentation.base.ErrorState
+
+class MapScreenModel(
+    private val order: ManageOrderUseCase,
+) : BaseScreenModel<MapScreenUiState, MapUiEffect>(MapScreenUiState()),
+    MapScreenInteractionListener {
+
+    init {
+        findingNewOrder()
+    }
+
+    private fun findingNewOrder() {
+        tryToExecute(
+            function = order::foundNewOrder,
+            onSuccess = ::onFoundNewOrderSuccess,
+            onError = ::onFoundNewOrderFailed
+        )
+    }
+
+    private fun onFoundNewOrderSuccess(order: Order) {
+        updateState {
+            it.copy(
+                isLoading = false,
+                isNewOrderFound = true,
+                error = null,
+                orderInfoUiState = order.toUiState()
+            )
+        }
+    }
+
+    private fun onFoundNewOrderFailed(errorState: ErrorState) {
+        updateState {
+            it.copy(
+                isLoading = false,
+                isNewOrderFound = false,
+                error = errorState,
+            )
+        }
+    }
+
+    override fun onClickAccept() {
+        updateState {
+            it.copy(
+                isLoading = false,
+                error = null,
+                isNewOrderFound = false,
+                isAcceptedOrder = true,
+            )
+        }
+    }
+
+    override fun onClickCancel() {
+        updateState {
+            it.copy(
+                isLoading = true,
+                isNewOrderFound = false,
+                isAcceptedOrder = false,
+                error = null
+            )
+        }
+        findingNewOrder()
+    }
+
+    override fun onClickArrived() {
+        updateState {
+            it.copy(
+                isLoading = false,
+                error = null,
+                orderInfoUiState = it.orderInfoUiState.copy(
+                    isArrived = true
+                )
+            )
+        }
+    }
```

Comment:
same point here when user click on Arrived button we are only updating the uiState i think it should be an end point we need to call unless how the end user will know the taxi driver accepted the order

Reply:
I don't have the endpoint yet